### DISCUSSION TITLE
Scaling an XYChart showing contour data

### DISCUSSION THREAD

**Discussion by cloudpaul** [GSI-CS-CO/chart-fx#597] [General]
Hi,
    I'm using a ChartFX XYChart with a ContourDataSetRender to display [GEBCO](https://www.gebco.net/data_and_products/gridded_bathymetry_data/) bathymetry data.
This is latitude/longitude data with an elevation at each data point.
I want to apply a Mercator projection to the chart and was wondering about the best way to do that. 
I think I might need something like the NonLinearTimeAxis (from TimeAxisNonLinearSample) to scale the axes.
Does this seem like the right approach? If f not I would appreciate some guidance!

**Comment by wirew0rm** [comment]
Hey and welcome to chartfx,

As long as each axis only depends on one dimension of the data (which seems to be the case for the mercator projection), it should be possible to do it the same way the nonlinear time axis does it. More complex transformations would be difficult at the moment.

Feel free to also post some screenshots (and/or links to your code if it's open source) of your usage of chart-fx to our [user gallery](https://github.com/fair-acc/chart-fx/discussions/categories/show-and-tell), your usecase sounds interesting, I think you're the first one to use chartfx for geographic data, which i'm sure will have some interesting special cases :)

**Reply by cloudpaul** [reply]
Thank you

**Reply by cloudpaul** [reply]
Unfortunately, the dimensions aren't independent. Do you have any other ideas as to how I might scale the chart?

**Reply by wirew0rm** [reply]
You can always perform the transformation before you put the data into your dataset. Note that since your data is on a grid in mercator coordinates and the contour renderer expects the data to be on a cartesian grid, you'll have to implement some sort of interpolation.

If you'd like to work on integrating better support for other coordinate systems into ChartFx it would be a welcome improvement albeit not a trivial one as it would require structural changes to the axes, renderers and zoom interactions at the very least.
